api/src/main/java/org/commonjava/indy/data/DefaultStoreValidator.java

```diff
@@ -0,0 +1,222 @@
+            errors.put(StoreValidationConstants.GENERAL, e.getMessage());
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl( remoteUrl.get().toExternalForm() )
+                .setErrors(errors)
+                .build();
+        }
+        if (artifactStore instanceof RemoteRepository) {
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setRepositoryUrl(((RemoteRepository) artifactStore).getUrl())
+                .setErrors(errors)
+                .build();
+        }
+        else {
+            return new ArtifactStoreValidateData
+                .Builder(artifactStore.getKey())
+                .setValid(true)
+                .setErrors(errors)
+                .build();
+        }
+    }
+
+    private boolean allowedNonSSLHostname(String remoteHost, String host) {
+        String[] remoteHostPartsTrimed =
+                                        Arrays.asList(remoteHost.split("\\."))
+                                            .stream()
+                                            .map(val -> val.trim())
+                                            .collect(Collectors.toList())
+                                            .toArray(new String[0]);
+        int remoteHostPartsLength = remoteHostPartsTrimed.length;
+        String[] hostParts = host.split("\\.");
+        int hostPartsLength = hostParts.length;
+        int i = 1;
+        int iter = remoteHostPartsLength;
+        while (iter > 0) {
+            String partRemoteHost = remoteHostPartsTrimed[remoteHostPartsLength - i];
+
+            if(i > (hostPartsLength-1)) {
+                return true;
+            }
+            String partHost = hostParts[hostPartsLength - i];
+
+            if(partRemoteHost.equalsIgnoreCase(partHost)
+                || (partRemoteHost.equalsIgnoreCase("*") || partRemoteHost.equals("") )) {
+                iter--;
+            } else {
+                return false;
+            }
+            i++;
+        }
+        return true;
+    }
+
+    private Future<Integer> executeGetHttp(HttpGet httpGetTask , CountDownLatch countDownLatch) {
+        countDownLatch.countDown();
+        return executorService.submit(() -> {
+            CloseableHttpClient closeableHttpClient = HttpClientBuilder.create().build();
+            try (CloseableHttpResponse response = closeableHttpClient.execute(httpGetTask)) {
+                LOGGER.warn("=> Check HTTP GET Response code: " + response.getStatusLine().getStatusCode());
+                return response.getStatusLine().getStatusCode();
+            } catch (IOException ioe) {
+                LOGGER.error(" => Not Successfull HTTP GET request from StoreValidatorRemote, \n => Exception: " + ioe);
+                throw new InvalidArtifactStoreException("Not valid remote Repository", ioe);
```

Comment:
Hmm, if we passed in the error map to this task, could we set the error value directly and avoid throwing an exception here? Assembling an exception requires getting the call stack IIRC, which is expensive.